planner/cascades/optimize.go

```diff
@@ -67,10 +67,21 @@ func (opt *Optimizer) GetImplementationRules(node plannercore.LogicalPlan) []Imp
 }
 
 // FindBestPlan is the optimization entrance of the cascades planner. The
-// optimization is composed of 2 phases: exploration and implementation.
+// optimization is composed of 3 phases: preprocessing, exploration and implementation.
 //
 //------------------------------------------------------------------------------
-// Phase 1: Exploration
+// Phase 1: Preprocessing
+//------------------------------------------------------------------------------
+//
+// The target of this phase is to preprocess the logical plan tree by some heuristic
+// rules which may traverse the whole plan tree.
```

Comment:
// The target of this phase is to preprocess the plan tree by some heuristic
// rules which should always be beneficial, for example `Column Pruning`.

Reply:
With this comment, we can remove the next comment line.

Reply:
addressed